### PR TITLE
#166240834 Users should be able to tag their articles

### PR DESCRIPTION
**Pull request by nkpremices**
[use case document](https://docs.google.com/document/d/1qp06cKR0aVI2rrdVG3jZNS8lTLmyfLZA8F24D6ex558/edit?usp=sharing)

#### What does this PR do?
Have the `Tag articles` endpoint working 

#### Description of Task to be completed?
* Document the endpoint using swagger
* Implement the feature logic
* write tests for the endpoint

#### How should this be manually tested?
After cloning the repo, follow the setup intructions, then use the swagger documentation to add a tag to an article.

#### Screenshots (if appropriate)
Screen Shot 2019-06-25 at 21 22 33

### CODE REVIEW

**Comment by MCFrank16** [comment]
be more specific on the way you document your PR

File: src/api/controllers/articleTagController.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import models from '../models/index';
+import status from '../../helpers/constants/status.codes';
+import sendError from '../../helpers/error.sender';
+import errorMessages from '../../helpers/constants/error.messages';
+// import errorMessages from '../../helpers/constants/error.messages';
```
Comment (MCFrank16) [inline src/api/controllers/articleTagController.js]: Ehhh @nkpremices remove this comment codes please

**Review by MCFrank16 — CHANGES_REQUESTED** [review]

File: src/api/controllers/articleTagController.js (hunk 1 above)
Comment (premlogic) [reply src/api/controllers/articleTagController.js]: Cool bro

**Review by MCFrank16 — APPROVED** [review]

**Review by Ramadhan0 — APPROVED** [review]

File: src/middlewares/checkArticleOwnership.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+const { Article } = models;
+/**
+ * containing all user's model controllers (signup, login)
+ *
+ * @export
+ * @class CheckUserOwnership
+ */
+class CheckUserOwnership {
+  /**
+   * this is a middleware which checks if the user is registered in to our database.
```
Comment (alainburindi) [inline src/middlewares/checkArticleOwnership.js:13]: it checks ownership or if the user is registered??

File: tests/article.tags.test.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      .patch('/api/articles/How-to-create-sequalize-seedss/tags')
+      .set('Authorization', `Bearer ${testUserToken}`)
+      .send({
+        tag: 'Musics'
+      })
+      .end((err, res) => {
+        res.should.have.status(status.NOT_FOUND);
+        done();
+      });
+  });
```
Comment (alainburindi) [inline tests/article.tags.test.js:87]: can you test a multi tags request??

**Review by alainburindi — CHANGES_REQUESTED** [review]

File: src/helpers/constants/validation.schemas.js [hunk 4]
```diff
@@ -11,15 +11,26 @@ const email = string
   .regex(/^[a-z._\-0-9]*[@][A-Za-z]*[.][a-z]{2,4}$/)
   .required();
 
+const THREE = 3;
+const FIVE = 5;
```
Comment (MemunaHaruna) [inline src/helpers/constants/validation.schemas.js:15]: @nkpremices better variable names needed

Comment (premlogic) [reply src/helpers/constants/validation.schemas.js:15]: noted @MemunaHaruna

**Pull request closed without merging**